ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/DatabaseSelfServiceError.scala

```diff
@@ -36,6 +36,8 @@ object DatabaseSelfServiceError {
     case ex: JdbcSQLIntegrityConstraintViolationException => retryable(ex)
     case ex: SQLRecoverableException => retryable(ex)
```

Comment:
I believe it'd be easier to manage `DatabaseSelfServiceError` logic if we split it per backend basis on the top level